pkg/connector/users.go

```diff
@@ -82,8 +97,15 @@ func (o *userBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken
 	return nil, "", nil, nil
 }
 
+func (o *userBuilder) getUsers() map[string]*client.User {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	return o.users
+}
```

Comment:
I don't think this is the right approach, since the sync could resume, we will receive zero users in workspace

Reply:
I had similar issue before

fix it with

https://github.com/ConductorOne/baton-sendgrid/blob/main/pkg/connector/scopes_cache.go